### PR TITLE
docs: HUGR spec copyediting

### PR DESCRIPTION
**Pull request by qartik**
- docs: fix hugr URLs
- docs(spec): normalize gremlins
- fix: prefer DOI/handle links over others for papers
- style: markdownlint it
- fix(spec): several edits

### CODE REVIEW

File: specification/hugr.md [hunk 1]
```diff
@@ -211,7 +214,7 @@ always local, i.e. source and target have the same parent.
 
 Here we define some core types of operation required to represent
 full programs, including dataflow operations (in
-[functions](#functions)).
+[functions](#dataflow)).
```
Comment (cqc-alec) [inline specification/hugr.md:217]: Should the link be attached to "dataflow" rather than "functions"?

File: specification/hugr.md [hunk 2]
```diff
@@ -769,17 +772,18 @@ indices after the list of node indices?
...
 
-  - The next class is ``CopyableType``, i.e. types holding ordinary classical
-    data, where values can be copied (and discarded, the 0-ary copy). This
-    allows multiple (or 0) outgoing edges from an outport; also these types can
-    be sent down Static edges. Note: dataflow inputs (Value and Static) always
-    require a single connection.
+- The next class is ``CopyableType``, i.e. types holding ordinary classical
+  data, where values can be copied (and discarded, the 0-ary copy). This
+  allows multiple (or 0) outgoing edges from an outport; also these types can
+  be sent down `Static` edges. Note: dataflow inputs (`Value` and `Static`) always
```
Comment (cqc-alec) [inline specification/hugr.md:783]: I wonder if we can standardize on single backticks throughout instead of double backticks? I think the only reason to use double backticks is if the text contains single-backtick characters (which are then rendered verbatim), but in most (all?) case here that doesn't apply. But maybe this would be best done in a separate PR.

**Review by cqc-alec — COMMENTED** [review]
Thanks! Just a few comments.

File: specification/hugr.md [hunk 3]
```diff
@@ -24,61 +24,62 @@ represent (typed) data or control dependencies.
...
+  have) only operations supported on specific targets are used.
+- A space efficient and user friendly specification of a subgraph and
+  replacement graph, along with an efficient routine for performing
+  the replacement.
 
 ## Functional description
 
-A HUGR is a directed graph. There are several different types of node, and
-several different types of edge, with different semantics, described below.
+A HUGR is a directed graph. There are several different types of nodes, and
```
[the remark this answers is not in the document]
Comment (qartik) [reply specification/hugr.md]: There is a lot of discussion on this at https://english.stackexchange.com/questions/29843/types-of-followed-by-singular-or-plural but since it's a matter of preference, I have reverted my changes.

File: specification/hugr.md [hunk 4]
```diff
@@ -138,7 +141,7 @@ As well as the type, dataflow edges are also parametrized by a
 `Locality`, which declares whether the edge crosses levels in the hierarchy. See
 [Edge Locality](#edge-locality) for details.
 
-```
+```haskell
```
[the remark this answers is not in the document]
Comment (qartik) [reply specification/hugr.md:144]: It's not. But I left it for consistency with other BNF-like code blocks in the doc for syntax highlighting.

File: specification/hugr.md (hunk 2 above)
Comment (qartik) [reply specification/hugr.md:782]: Fixed.

**Review by cqc-alec — APPROVED** [review]

File: specification/hugr.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (cqc-alec) [reply specification/hugr.md]: I wonder if there is a difference between American and British English here. I was interested in [this comment](https://english.stackexchange.com/a/5546) (from a linked page):

> I think that the construction of the form "two types of apple" sounds more than stilted; it just plain sounds awkward, and I would be surprised that it sounds familiar and normal to anyone (at least speaking for US English).

To me it sounds perfectly natural!

[the remark this answers is not in the document]
Comment (qartik) [reply specification/hugr.md]: Most likely, as I have the same experience as that commenter.
